Fixed missed issues in #1806 as part of ADDREV-14 (#1820)
mozilla/extension-workshop: src/content/documentation/develop/best-practices-for-collecting-user-data-consents.md

In this example:

1. Your extension should start up with all data collection functionality disabled.

> 2. Immediately after installation (see [Best practices for onboarding, upboarding, and offboarding users](/documentation/develop/onboard-upboard-offboard-users/)), present them with a consent dialog.

Reviewer: The current construction of this sentence splits the concept being conveyed with a parenthetical. This makes the sentence more difficult for a reader to follow than necessary.

```suggestion
2. Immediately after installation, present them with a consent dialog. (See [Best practices for onboarding, upboarding, and offboarding users](/documentation/develop/onboard-upboard-offboard-users/) for more.)
```

   On this dialog:
3. The option for collecting personally identifying information must default to disabled so that the user has to actively opt-in.
4. The options for technical data may default to enabled, so that the user may choose to opt-out.
5. If the user:
    - accepts mandatory and optional data collection &rarr; continue with data collection enabled enabled.

>     - declines mandatory data collection &rarr; uninstall the extension with [`management.uninstallSelf()`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/management/uninstallSelf).
>     - declines optional data collection &rarr; disable optional data collection and continue running the extension.

Reviewer: Swap the order of these bullets so the sequence of choices appears in order of desirability for the developer.
```suggestion
    - declines optional data collection &rarr; disable optional data collection and continue running the extension.
    - declines mandatory data collection &rarr; uninstall the extension with [`management.uninstallSelf()`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/management/uninstallSelf).
```

In this example:

1. Your extension should start up with all data collection functionality disabled.
2. Immediately after installation (see [Best practices for onboarding, upboarding, and offboarding users](/documentation/develop/onboard-upboard-offboard-users/)), present them with a consent dialog.
   On this dialog:
3. The option for collecting personally identifying information must default to disabled so that the user has to actively opt-in.
4. The options for technical data may default to enabled, so that the user may choose to opt-out.
5. If the user:
    - opts IN to all features &rarr; continue with all extension features enabled.

>     - opts OUT of any feature that cannot be disabled &rarr; uninstall the extension with [`management.uninstallSelf()`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/management/uninstallSelf).

Reviewer 2: This doesn't sound right. How can you opt out of something that cannot disabled?

Author: I see, and yes that's fair. It could be made more explicit from the perspective of what a user can do.

I'll make it to be something like, if that sounds more in line with what we mean.

```
If the user:
- consents to all features ...
- doesn't consent to any feature that cannot be disabled ...
- doesn't consent to any feature that can be disabled ...
```

Any thoughts? I'll update diagram accordingly.

Author: @wagnerand made the changes (to both diagram and accompanying description text) to sound more explicit from perspective of user action performed, and used "data collection" instead of "features" to be more explicit. Please take a look! Thanks

Once you understand your privacy consent flow, you can add a suitable consent dialog. The following are suggestions on how to present a data collection consent dialog. We recommend that all of the data collection controls and actions (toggles and confirmation buttons) be present within the same view. A summary of the data collected and how it is used must be present for each type of data collected within the consent dialog.

> Where the mockups include the linked text “our Privacy Policy”, this should go to your extension’s privacy policy page. Privacy policy page can be self-hosted for unlisted extensions or hosted on AMO for listed extensions.

Reviewer 2: Turn the _can_ into  a _should_.

  ![Mockup of a prompt that could be used when extension requires consent for processing personal and technical data.](/assets/img/documentation/develop/consent-mixed.jpg)
### Required data collection

> The extension requires personal or technical data collection to provide its main functionality and cannot be used without. The consent decline option uninstalls the extension using [`management.uninstallSelf()`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/management/uninstallSelf).

Reviewer: Are extensions required to uninstall themselves if the user opts out of required data collection? If not, we may want to clarify that here and in other parts of this document.

```suggestion
The extension requires personal or technical data collection to provide its main functionality and cannot be used without this data. The consent decline option should either disable all data collection or uninstall the extension using [`management.uninstallSelf()`](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/management/uninstallSelf).
```

Author: > Are extensions required to uninstall themselves if the user opts out of required data collection
Yes, we require them to do that. Since the pretext to this is that the extension requires the said data to provide its main functionality, the assumption is that it cannot be disabled and warrants an uninstall
